### ISSUE TITLE
Athena: SQLFluff doesn't show error in specific place when using CTE `with` construction in SQL query

### ISSUE THREAD

**Issue by sanromeo** [sqlfluff/sqlfluff#5433]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff does not show an error in a specific place when using CTE with a construction in a query

### Expected Behaviour

sqlfluff lint show correct place where I've put two commas in a row

I expect to see something like that
L:   5 | P:  2 | <rule> | Found One extra comma ','. Expected only one.

### Observed Behaviour

sqlfluff lint show unparsable section when I put two commas in a row between CTE
```
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: "with
                       | query_a as (\nselect\n    'John' as n..."
WARNING: Parsing errors found and dialect is set to 'athena'. Have you configured your dialect correctly?
==== summary ====
violations:        1 status:         FAIL
All Finished 📜 🎉!
```

### How to reproduce

sql query
```sql
with query_a as (
select
    'John' as name,
    'Doe' as surname
),

, query_b as (
select
    'Jess' as name,
    'Brown' as surname
)

select * from query_a
```

```
sqlfluff lint -v --processes 4 --config <path-to-config> <path-to-model>
```

### Dialect

Athena

### Version

sqlfluff, version 2.3.5

### Configuration

.sqlfluff
```yaml
[sqlfluff]
templater = dbt
dialect = athena
large_file_skip_char_limit = 0
large_file_skip_byte_limit = 0

exclude_rules = AL05, AL07, AM02, AM05, CV10, LT02, RF02, ST06, ST07, RF01, RF03, LT01
warnings = RF01
max_line_length = 120

[sqlfluff:templater:dbt]
profiles_dir = ./
target = athena

[sqlfluff:indentation]
tab_space_size = 4
indent_unit = space
indented_joins = false
indented_using_on = true
template_blocks_indent = false
indented_ctes = false
allow_implicit_indents = true

[sqlfluff:layout:type:comma]
line_position = leading

[sqlfluff:rules:references.keywords]
ignore_words = comment,data,date,domain,error,first,is_external,login,operator,schema,source,timestamp

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = lower
```

.sqlfluffignore
```yaml
.venv/*
macros/*
dbt_packages/*
target/
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
